src/vaults/curve/CurveERC4626.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.10;
+
+import {ERC20, ERC4626} from "solmate/mixins/ERC4626.sol";
+import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
+
+import {RewardsClaimer} from "../../utils/RewardsClaimer.sol";
+
+interface ILiquidityGauge {
+    function deposit(uint256 value) external;
+
+    function withdraw(uint256 value, bool claim_rewards) external;
+
+    function claim_rewards() external;
+
+    function balanceOf(address) external view returns (uint256);
+
+    function staking_token() external view returns (address);
+
+    function reward_tokens(uint256 i) external view returns (address token);
+
+    function reward_count() external view returns (uint256 nTokens);
+}
+
+/// @title Curve Finance Yield Bearing Vault
+/// @author joeysantoro
+contract CurveERC4626 is ERC4626, RewardsClaimer {
+    using SafeTransferLib for ERC20;
+
+    /// @notice The Convex Rewards contract (for claiming rewards)
+    ILiquidityGauge public immutable gauge;
+
+    /**
+     @notice Creates a new Vault that accepts a specific underlying token.
+     @param _asset The ERC20 compliant token the Vault should accept.
+     @param _name The name for the vault token.
+     @param _symbol The symbol for the vault token.
+     @param _gauge The Convex Rewards contract (for claiming rewards).
+     @param _rewardsDestination the address to send CRV and CVX.
```

Comment:
not necessarily CRV and CVX, it can be BAL for Balancer gauges too. And CVX don't work with a gauge system, only Curve distributes CRV this way